lib/block-supports/shadow.php

```diff
@@ -0,0 +1,80 @@
+<?php
+/**
+ * Shadow block support flag.
+ *
+ * @package gutenberg
+ */
+
+/**
+ * Registers the style and shadow block attributes for block types that support it.
+ *
+ * @param WP_Block_Type $block_type Block Type.
+ */
+function gutenberg_register_shadow_support( $block_type ) {
+	$has_shadow_support = block_has_support( $block_type, array( 'shadow' ), false );
+
+	if ( ! $has_shadow_support ) {
+		return;
+	}
+
+	if ( ! $block_type->attributes ) {
+		$block_type->attributes = array();
+	}
+
+	if ( $has_shadow_support && ! array_key_exists( 'style', $block_type->attributes ) ) {
+		$block_type->attributes['style'] = array(
+			'type' => 'object',
+		);
+	}
+
+	if ( $has_shadow_support && ! array_key_exists( 'shadow', $block_type->attributes ) ) {
+		$block_type->attributes['shadow'] = array(
+			'type' => 'string',
+		);
+	}
+}
+
+/**
+ * Add CSS classes and inline styles for shadow features to the incoming attributes array.
+ * This will be applied to the block markup in
+ * the front-end.
+ *
+ * @param  WP_Block_Type $block_type       Block type.
+ * @param  array         $block_attributes Block attributes.
+ *
+ * @return array Shadow CSS classes and inline styles.
+ */
+function gutenberg_apply_shadow_support( $block_type, $block_attributes ) {
+	if ( ! property_exists( $block_type, 'supports' ) ) {
+		return array();
+	}
+
+	$has_shadow_support = _wp_array_get( $block_type->supports, array( 'shadow' ), false );
```

Comment:
Can we use `block_has_support` here too?

Reply:
Yep, missed that, done.